plugins/crypto/mbedtls/ua_pki_mbedtls.c

```diff
@@ -471,6 +475,35 @@ certificateVerification_verify(void *verificationContext,
         } else {
             retval = UA_STATUSCODE_BADSECURITYCHECKSFAILED;
         }
+
+#ifdef UA_ENABLE_CERT_REJECTED_DIR
+        if(ci->rejectedListFolder.length !=0) {
+            char rejectedFileName[256] = {0};
+            UA_ByteString thumbprint;
+            UA_ByteString_allocBuffer(&thumbprint, UA_SHA1_LENGTH);
+            if(mbedtls_thumbprint_sha1(certificate, &thumbprint) == UA_STATUSCODE_GOOD) {
+                static const char hex2char[] = "0123456789ABCDEF";
+                for (size_t pos = 0, namePos = 0; pos < thumbprint.length; pos++) {
+                    rejectedFileName[namePos++] = hex2char[(thumbprint.data[pos] & 0xf0) >> 4];
+                    rejectedFileName[namePos++] = hex2char[thumbprint.data[pos] & 0x0f];
+                }
+                strcat(rejectedFileName, ".der");
+            } else {
+                UA_UInt32 dt = (UA_UInt32)UA_DateTime_now();
```

Comment:
UA_DateTime_now returns the 64 bit signed integer which represents the number of 100 nanosecond intervals since January 1, 1601. If you cast this value to a 32 bit unsigned integer, you get the same value every 7,15 minutes. Should the UA_DateTime_now be converted to seconds with UA_DATETIME_SEC or should the 64-bit integer be used for the file name?

Reply:
I changed it to use 64-bit unsigned integer. I'm not sure if we should save the rejected certificate in this case at all.